openreview/conference/builder.py

```diff
@@ -1540,6 +1547,63 @@ def is_release_authors(is_note_accepted):
             self.set_homepage_decisions(decision_heading_map=decision_heading_map)
         self.client.remove_members_from_group('active_venues', self.id)
 
+    def post_decisions(self, decisions_file):
+        decisions_data = list(csv.reader(StringIO(decisions_file.decode()), delimiter=","))
+        decision_notes = {
+            n.forum: n for n in self.client.get_all_notes(
+                invitation=self.get_invitation_id(self.decision_stage.name, '.*')
+            )}
+        paper_notes = {n.forum: n for n in self.get_submissions()}
+
+        def post_decision(paper_decision):
+            if len(paper_decision) < 2:
+                raise OpenReviewException(
+                    "Not enough values provided in the decision file. Expected values are: paper_id, decision, comment")
+            if len(paper_decision) > 3:
+                raise OpenReviewException(
+                    "Too many values provided in the decision file. Expected values are: paper_id, decision, comment"
+                )
+            if len(paper_decision) == 3:
+                paper_id, decision, comment = paper_decision
+            else:
+                paper_id, decision = paper_decision
+                comment = ''
+
+            print(f"Posting Decision {decision} for Paper {paper_id}")
+            paper_note = paper_notes.get(paper_id, None)
+            if not paper_note:
+                raise OpenReviewException(
+                    f"Paper with ID: {paper_id} not found. Please check the submitted paperIDs."
+                )
+
+            paper_decision_note = decision_notes.get(paper_id, None)
+            if paper_decision_note:
+                paper_decision_note.content = {
+                    'title': 'Paper Decision',
+                    'decision': decision.strip(),
+                    'comment': comment,
+                }
+            else:
+                paper_decision_note = openreview.Note(
+                    invitation=self.get_invitation_id(name=self.decision_stage.name, number=paper_note.number),
+                    writers=[self.get_program_chairs_id()],
+                    readers=self.decision_stage.get_readers(conference=self, number=paper_note.number),
+                    nonreaders=self.decision_stage.get_nonreaders(conference=self, number=paper_note.number),
+                    signatures=[self.get_program_chairs_id()],
+                    content={
+                        'title': 'Paper Decision',
+                        'decision': decision.strip(),
```

Comment:
I would also check if the decision is part of the decision options of the decision stage and report this in the decision status note.